dotnet/src/SemanticKernel/TemplateEngine/PromptTemplateEngine.cs

```diff
@@ -65,17 +65,17 @@ public async Task<string> RenderAsync(string templateText, SKContext context)
     public async Task<string> RenderAsync(IList<Block> blocks, SKContext context)
     {
         this._log.LogTrace("Rendering list of {0} blocks", blocks.Count);
-        var result = new StringBuilder();
+        var tasks = new List<Task<string>>();
```

Comment:
May as well pass in the count to reserve the space, since we know exactly how many there will be!
```suggestion
        var tasks = new List<Task<string>>(blocks.Count);
```